### PR TITLE
Add mlr3pipelines framework comparison to Appendix

### PR DESCRIPTION
**Pull request by pfistfl**

### CODE REVIEW

File: bookdown/98-appendix.Rmd [hunk 1]
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
...
+space of a `Pipeline` more efficient, as executing some steps multiple times can be avoided.
+
+We compare functionality available in both `mlr3pipelines` and `sklearn.pipeline.Pipeline` to give a
+comparison.
+
+
+
+The following example obtained from the **sklearn** documentation showcases a **Pipeline**
+that first Selects a feature and performs PCA on the original data, concatenates the resulting datasets
+and applies a Support Vector Machine.
```
Comment (pat-s) [inline bookdown/98-appendix.Rmd]: - watch package linking
- watch linebreaking

File: bookdown/98-appendix.Rmd [hunk 2]
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
...
+tuner = TunerGridSearch$new(pe, terminator, resolution = 10)$tune()
+
+# Set the learner to the optimal values and train
+learner$param_set$values = tuner$tune_result()$values
+```
+
+In summary, we can achieve similar results with a comparable number of lines, while at the
+same time offering greater flexibility with respect to which kinds of pipelines we want to
+optimize over. At the same time, experiments using `mlr3` can now be arbitrarily parallelized
+using `futures`.
```
Comment (pat-s) [inline bookdown/98-appendix.Rmd]: - watch linebreaking
- watch package linking

File: bookdown/98-appendix.Rmd [hunk 3]
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
...
+
+### mlr3pipelines vs recipes
+
+**recipes** is a new package, that covers some of the same applications steps as
+**mlr3pipelines**.
+Both packages feature the possiblility to connect different pre- and post-processing
+methods using a pipe-operator.
+As `recipes` package tightly integrates with the `tidymodels` ecosystem, much of the
+functionality integrated there can be used in `recipes`.
+We compare `recipes` to `mlr3pipelines` using an example from the `recipes` vignette.
```
Comment (pat-s) [inline bookdown/98-appendix.Rmd]: - watch linebreaking
- watch package linking

File: bookdown/98-appendix.Rmd [hunk 4]
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
...
+
+The aim of the analysis is to predict whether customers pay back their loans given some
+information on the customers. In order to do this, we build a model that does the following:
+1. It first imputes missing values using k-nearest neighbours
+2. All factor variables are converted to numerics using dummy encoding
+3. The data is first centered then scaled.
```
Comment (pat-s) [inline bookdown/98-appendix.Rmd]: - watch linebreaking
- list should have an empty line before

File: bookdown/98-appendix.Rmd [hunk 5]
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
...
+
+In order to validate the algorithm, data is first split into a train and test set using
+`initial_split`, `training`, `testing`. The recipe trained on the train data (see steps above)
+is then applied to the test data.
```
Comment (pat-s) [inline bookdown/98-appendix.Rmd]: watch linebreaking

**Comment by pfistfl** [comment]
I think I have everything.

- Added mlr3pipelines table. I think this is better than nothing for now
- Repaired the vignette to concur with style-guide
- Added a few lines to the pipeline's section in order to introduce sugar appropriately
